test/sql/catalog/function/test_recursive_macro.test

```diff
@@ -39,13 +39,10 @@ create macro m1(a) as a+1;
 statement ok
 create macro m2(a) as m1(a)+1;
 
-statement ok
-create or replace macro m1(a) as m2(a)+1;
-
 statement error
-select m2(42);
+create or replace macro m1(a) as m2(a)+1;
 ----
-Binder Error: Maximum recursion depth exceeded
+Catalog Error: CREATE OR REPLACE is not allowed to depend on itself
```

Comment:
This showcases the issue for CREATE OR REPLACE
I thought this would get triggered by `create or replace table t1 as select * from t1` but it won't